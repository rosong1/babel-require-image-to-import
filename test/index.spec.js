const plugin = require('../src');
const babel = require('@babel/core');

it('simple', () => {
  const example = `
const img1 = require('./img/icon1.png');
`
  const { code } = babel.transform(example, { plugins: [plugin] })
  expect(code).toMatchSnapshot();
});

it('with array', () => {
  const example = `
const img1 = require('./img/icon1.png');

const list = [
  {
    label: "item1",
    value: require("@/img/item1.jpg"),
  },
  {
    label: "item2",
    value: require("@/img/item2.svg"),
  },
];
`
  const { code } = babel.transform(example, { plugins: [plugin] })
  expect(code).toMatchSnapshot();
});

it('with jsx', () => {
  const example = `
  const img1 = require("./img/icon1.webp");

  const list = [
    {
      label: "item1",
      value: require("@/img/item1.jpeg"),
    },
    {
      label: "item2",
      value: require("@/img/item2.png"),
    },
  ];
  
  const ImgComponent = (props) => {
    const { onError, defaultImg, ...rest } = props;
    return (
      <img
        {...rest}
        onError={(e) => {
          e.target.onerror = null;
          e.target.src = defaultImg || require("./img/default-avatar.png");
        }}
      />
    );
  }; 
`
  const { code } = babel.transform(example, { presets: ["@babel/preset-react"], plugins: [plugin] })
  expect(code).toMatchSnapshot();
});