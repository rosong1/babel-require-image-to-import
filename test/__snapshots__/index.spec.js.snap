// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple 1`] = `
"import _img_uid from \\"./img/icon1.png\\";
const img1 = _img_uid;"
`;

exports[`with array 1`] = `
"import _img_uid3 from \\"@/img/item2.svg\\";
import _img_uid2 from \\"@/img/item1.jpg\\";
import _img_uid from \\"./img/icon1.png\\";
const img1 = _img_uid;
const list = [{
  label: \\"item1\\",
  value: _img_uid2
}, {
  label: \\"item2\\",
  value: _img_uid3
}];"
`;

exports[`with jsx 1`] = `
"function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import _img_uid4 from \\"./img/default-avatar.png\\";
import _img_uid3 from \\"@/img/item2.png\\";
import _img_uid2 from \\"@/img/item1.jpeg\\";
import _img_uid from \\"./img/icon1.webp\\";
const img1 = _img_uid;
const list = [{
  label: \\"item1\\",
  value: _img_uid2
}, {
  label: \\"item2\\",
  value: _img_uid3
}];

const ImgComponent = props => {
  const {
    onError,
    defaultImg,
    ...rest
  } = props;
  return /*#__PURE__*/React.createElement(\\"img\\", _extends({}, rest, {
    onError: e => {
      e.target.onerror = null;
      e.target.src = defaultImg || _img_uid4;
    }
  }));
};"
`;
